src/test/java/org/sagebionetworks/bridge/services/ParticipantFileServiceTest.java

```diff
@@ -0,0 +1,206 @@
+
+        verify(mockFileDao).getParticipantFiles("test_user", "dummy-key", 20);
+    }
+
+    @Test
+    public void getParticipantFilesNoOffsetKey() {
+        service.getParticipantFiles("test_user", null, 20);
+
+        verify(mockFileDao).getParticipantFiles("test_user", null, 20);
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getParticipantFilesPageSizeTooSmall() {
+        service.getParticipantFiles("test_user", null, 1);
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getParticipantFilesPageSizeTooLarge() {
+        service.getParticipantFiles("test_user", null, 5000);
+    }
+
+    @Test
+    public void getParticipantFile() {
+        String downloadUrl = "https://" + UPLOAD_BUCKET + "/test_user/file_id";
+        ParticipantFile file = ParticipantFile.create();
+        file.setFileId("file_id");
+        file.setUserId("test_user");
+        file.setAppId("api");
+        file.setMimeType("dummy-type");
+        file.setCreatedOn(TestConstants.TIMESTAMP);
+
+        when(mockFileDao.getParticipantFile("test_user", "file_id")).thenReturn(Optional.of(file));
+
+        ParticipantFile result = service.getParticipantFile("test_user", "file_id");
+        assertEquals(result.getUserId(), "test_user");
+        assertEquals(result.getFileId(), "file_id");
+        assertEquals(result.getCreatedOn(), TestConstants.TIMESTAMP);
+        assertEquals(result.getMimeType(), "dummy-type");
+        assertEquals(result.getDownloadUrl(), downloadUrl);
+        assertEquals(result.getAppId(), "api");
+
+        verify(mockS3Client).generatePresignedUrl(requestCaptor.capture());
+        GeneratePresignedUrlRequest request = requestCaptor.getValue();
+        assertEquals(request.getMethod(), HttpMethod.GET);
+        assertEquals(request.getContentType(), file.getMimeType());
+        assertEquals(request.getKey(), "test_user/file_id");
+        assertEquals(request.getRequestParameters().get(Headers.SERVER_SIDE_ENCRYPTION),
+                ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
+    }
+
+    @Test(expectedExceptions = EntityNotFoundException.class)
+    public void getParticipantFileNoSuchFile() {
+        when(mockFileDao.getParticipantFile(any(), any())).thenReturn(Optional.empty());
+
+        service.getParticipantFile("test_user", "file_id");
+    }
+
+    @Test
+    public void createParticipantFile() {
+        String upload = "https://" + UPLOAD_BUCKET + "/test_user/file_id";
+
+        when(mockFileDao.getParticipantFile(any(), any())).thenReturn(Optional.empty());
+
+        // UserId and AppId should not depend on file, it should be manually set by the Service.
+        ParticipantFile file = ParticipantFile.create();
+        file.setFileId("file_id");
+        file.setUserId("wrong_user");
+        file.setMimeType("dummy-type");
+        file.setAppId("wrong_api");
+        file.setCreatedOn(TestConstants.TIMESTAMP);
```

Comment:
Don't set the timestamp here, so we can verify that the service sets the timestamp.